src/xrpld/app/tx/detail/Batch.cpp

```diff
@@ -32,23 +33,69 @@ namespace ripple {
 XRPAmount
 Batch::calculateBaseFee(ReadView const& view, STTx const& tx)
 {
+    XRPAmount const maxAmount{std::numeric_limits<std::int64_t>::max()};
+
+    // batchBase: view.fees().base for batch processing + default base fee
+    XRPAmount const baseFee = Transactor::calculateBaseFee(view, tx);
+
+    // LCOV_EXCL_START
+    if (baseFee > maxAmount - view.fees().base)
+        return INITIAL_XRP;
+    // LCOV_EXCL_STOP
+
+    XRPAmount const batchBase = view.fees().base + baseFee;
+
     // Calculate the Inner Txn Fees
     XRPAmount txnFees{0};
     if (tx.isFieldPresent(sfRawTransactions))
     {
         auto const& txns = tx.getFieldArray(sfRawTransactions);
+
+        XRPL_ASSERT(
+            txns.size() <= maxBatchTxCount,
+            "Raw Transactions array exceeds max entries.");
+
+        // LCOV_EXCL_START
+        if (txns.size() > maxBatchTxCount)
+            return INITIAL_XRP;
```

Comment:
i prefer throwing an logic error than returning INITIAL_XRP in these places

Reply:
fixed: https://github.com/XRPLF/rippled/pull/5060/commits/91a54023b0a7a133f8d9ec2950c7dd386f4ee2ba